core/src/test/java/org/apache/iceberg/avro/TestAvroSchemaProjection.java

```diff
@@ -150,4 +150,58 @@ public void projectWithMapSchemaChanged() {
         .as("Result of buildAvroProjection is missing some IDs")
         .isFalse();
   }
+
+  @Test
+  public void projectWithVariantSchemaChanged() {
+    final org.apache.avro.Schema currentAvroSchema =
+        SchemaBuilder.record("myrecord")
+            .fields()
+            .name("f11")
```

Comment:
Nit: it is usually better to have reasonable names, even in test schemas. Why not use "id" and "payload" or something similar?